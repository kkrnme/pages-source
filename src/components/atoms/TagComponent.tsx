import React, { ReactChild } from "react"
import { Stylable } from "../../utils/Components"
import { Link } from "gatsby"
import { PlainComponent } from "../../utils/PlainComponent"
import styled from "@emotion/styled"
import { ds, ss, cs } from "../../styles"

type TagComponentProps = Stylable & {
  children: ReactChild
}

type TagLinkProps = TagComponentProps & { children: string }

const Plain: PlainComponent<TagComponentProps> = props => (
  <label className={props.className}>{props.children}</label>
)

export const TagComponent = styled(Plain)`
  margin-left: 0.5rem;
  ${ds.padx(".5rem")}
  ${ss.rounded}
  letter-spacing: 0;
  color: ${cs.background};
  border: ${ss.border};
`

export const TagLink: React.FC<TagLinkProps> = props => (
  <TagComponent className="bg-yellow-500">
    <Link to={`/tags/${props.children}/`}>{props.children}</Link>
  </TagComponent>
)
