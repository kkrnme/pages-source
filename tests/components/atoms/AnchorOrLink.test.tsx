import React from "react"
import renderer from "react-test-renderer"

import { AnchorOrLink } from "../../../src/components/atoms/AnchorOrLink"

describe("AnchorOrLink", () => {
  for (const i of [
    "hoge",
    "/hoge",
    "hoge/",
    "/hoge/",
    "hoge/fuga",
    "hoge/fuga/",
    "/hoge/fuga/",
    "google.com",
    "google.com/",
    "https://google.com",
    "https://google.com/",
  ]) {
    it(`renders correctly: ${i}`, () => {
      const tree = renderer
        .create(<AnchorOrLink to={i}>{i}</AnchorOrLink>)
        .toJSON()
      console.log(
        `${i} => ${
          Object.keys(tree?.props!).includes("aria-current")
            ? "Link - internal link"
            : "a - external link"
        }`
      )
      expect(tree).toMatchSnapshot()
    })
  }
})
