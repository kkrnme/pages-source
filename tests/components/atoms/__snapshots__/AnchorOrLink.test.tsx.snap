// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnchorOrLink renders correctly: /hoge 1`] = `
<a
  href="/hoge"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  /hoge
</a>
`;

exports[`AnchorOrLink renders correctly: /hoge/ 1`] = `
<a
  href="/hoge/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  /hoge/
</a>
`;

exports[`AnchorOrLink renders correctly: /hoge/fuga/ 1`] = `
<a
  href="/hoge/fuga/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  /hoge/fuga/
</a>
`;

exports[`AnchorOrLink renders correctly: google.com 1`] = `
<a
  href="google.com"
>
  google.com
</a>
`;

exports[`AnchorOrLink renders correctly: google.com/ 1`] = `
<a
  href="google.com/"
>
  google.com/
</a>
`;

exports[`AnchorOrLink renders correctly: hoge 1`] = `
<a
  href="hoge"
>
  hoge
</a>
`;

exports[`AnchorOrLink renders correctly: hoge/ 1`] = `
<a
  href="hoge/"
>
  hoge/
</a>
`;

exports[`AnchorOrLink renders correctly: hoge/fuga 1`] = `
<a
  href="hoge/fuga"
>
  hoge/fuga
</a>
`;

exports[`AnchorOrLink renders correctly: hoge/fuga/ 1`] = `
<a
  href="hoge/fuga/"
>
  hoge/fuga/
</a>
`;

exports[`AnchorOrLink renders correctly: https://google.com 1`] = `
<a
  href="https://google.com"
>
  https://google.com
</a>
`;

exports[`AnchorOrLink renders correctly: https://google.com/ 1`] = `
<a
  href="https://google.com/"
>
  https://google.com/
</a>
`;
